src/plugins/intel_cpu/src/nodes/subgraph.h

```diff
@@ -47,6 +46,17 @@ class Snippet : public Node {
 
     // if generator is set, it would execute generated code otherwise it would fallback to nGraph reference
     void execute(dnnl::stream strm) override;
+    void executeDynamicImpl(dnnl::stream strm) override;
+
+    struct SnippetAttrs {
+        // Local copy of subgraph node for canonization & code generation
+        std::shared_ptr<ngraph::snippets::op::Subgraph> snippet;
```

Comment:
Why do we need a deep `snippet` copy in every `SnippetsAttrs` instance? In order to perform lowering and code generation in `SnippetJitExecutor(...)` it's enough to have a local copy of the reshaped `snippet` (with a life scope inside the constructor).

Reply:
done, thank you Ivan!